### PR TITLE
mpvScripts.mpv-discord: init at 1.6.1

### LINKED ISSUE #330795: Package request: mpv-discord
**Project description**

A cross-platform Discord Rich Presence integration for mpv with no external dependencies. (MPV Userscript)

**Metadata**

* homepage URL: https://github.com/tnychn/mpv-discord
* source URL: https://github.com/tnychn/mpv-discord
* license: mit
* platforms: unix, linux, darwin, windows

### PR DESCRIPTION
**Pull request by bddvlpr**
## Description of changes
Closes #330795

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [X] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [X] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/video/mpv/scripts/mpv-discord.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+buildLua {
+  pname = "mpv-discord";
+  inherit version src;
+
+  scriptPath = "scripts/discord.lua";
+
+  postInstall = ''
+    substituteInPlace $out/share/mpv/scripts/discord.lua \
+      --replace-fail 'binary_path = ""' 'binary_path = "${core}/bin/mpv-discord"' \
+      --replace-warn 'version = "1.4.1"' 'version = "${version}"'
```
Comment (eclairevoyant) [inline pkgs/applications/video/mpv/scripts/mpv-discord.nix:35]:
```suggestion
      --replace-fail 'version = "1.4.1"' 'version = "${version}"'
```

**Comment by eclairevoyant** [comment]
Please update the commit message to `mpvScripts.mpv-discord: init at 1.6.1`

**Comment by eclairevoyant** [comment]
@ofborg build mpvScripts.mpv-discord

File: pkgs/applications/video/mpv/scripts/mpv-discord.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  lib,
+}:
+let
+  version = "1.6.1";
+
+  src = fetchFromGitHub {
+    owner = "tnychn";
+    repo = "mpv-discord";
+    rev = "v${version}";
+    hash = "sha256-Uvm4McZR88s9tsstqLz7O2F3KkqsGlGrDoL8RjgQnVI=";
```
Comment (Artturin) [inline pkgs/applications/video/mpv/scripts/mpv-discord.nix:14]:
```suggestion
    hash = "sha256-P1UaXGboOiqrXapfLzJI6IT3esNtflkQkcNXt4Umukc=";
```

Remember to check CI failures.

**Pull request merged**